### PR TITLE
Move Circle Gizmos to Their Own File

### PR DESCRIPTION
**Pull request by SIGSTACKFAULT**
## Objective

- Give all the intuitive groups of gizmos their own file
- don't be a breaking change
- don't change Gizmos interface
- eventually do arcs too
- future types of gizmos should be in their own files
- see also https://github.com/bevyengine/bevy/issues/9400

## Solution

- Moved `gizmos.circle`, `gizmos.2d_circle`, and assorted helpers into `circles.rs`
- Can also do arcs in this MR if y'all want; just figured I should do one thing at a time.

## Changelog

- Changed
  - `gizmos::CircleBuilder` moved to `gizmos::circles::Circle2dBuilder`
  - `gizmos::Circle2dBuilder` moved to `gizmos::circles::Circle2dBuilder`

## Migration Guide

- change `gizmos::CircleBuilder` to `gizmos::circles::Circle2dBuilder`
- change `gizmos::Circle2dBuilder` to `gizmos::circles::Circle2dBuilder`

### CODE REVIEW

File: crates/bevy_gizmos/src/circles.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
+//! Additional [`Gizmo`] Functions -- Arrows
+
+use crate::prelude::Gizmos;
+use bevy_math::{Quat, Vec2, Vec3};
+use bevy_render::color::Color;
+use std::f32::consts::TAU;
+
+pub const DEFAULT_CIRCLE_SEGMENTS: usize = 32;
+
+pub fn circle_inner(radius: f32, segments: usize) -> impl Iterator<Item = Vec2> {
```
Comment (alice-i-cecile) [inline crates/bevy_gizmos/src/circles.rs:10]: This was made pub, but should probably stay private.

File: crates/bevy_gizmos/src/circles.rs [hunk 2]
```diff
@@ -0,0 +1,142 @@
+//! Additional [`Gizmo`] Functions -- Arrows
+
+use crate::prelude::Gizmos;
+use bevy_math::{Quat, Vec2, Vec3};
+use bevy_render::color::Color;
+use std::f32::consts::TAU;
+
+pub const DEFAULT_CIRCLE_SEGMENTS: usize = 32;
```
Comment (alice-i-cecile) [inline crates/bevy_gizmos/src/circles.rs:8]: I'm fine with this getting exposed in this PR, but if so, doc strings.

File: crates/bevy_gizmos/src/circles.rs (hunk 1 above)
Comment (alice-i-cecile) [reply crates/bevy_gizmos/src/circles.rs:10]: If you'd prefer it to be pub, it needs a better name and doc strings :)

File: crates/bevy_gizmos/src/lib.rs [hunk 3]
```diff
@@ -17,6 +17,7 @@
 //! See the documentation on [`Gizmos`] for more examples.
 
 mod arrows;
+mod circles;
```
Comment (alice-i-cecile) [inline crates/bevy_gizmos/src/lib.rs:20]: This should probably be pub? Along with `arrows`. Currently I don't see any way to access these types externally.

**Comment by alice-i-cecile** [comment]
Currently this is a breaking change, since we're not re-exporting.

File: crates/bevy_gizmos/src/circles.rs (hunk 1 above)
Comment (SIGSTACKFAULT) [reply crates/bevy_gizmos/src/circles.rs:10]: Yeah i thought it was pub before. just reverted it; that can be a different PR.

Comment (SIGSTACKFAULT) [reply crates/bevy_gizmos/src/circles.rs:10]: Heck i thought this was about DEFAULT_CIRCLE_SEGMENTS

File: crates/bevy_gizmos/src/circles.rs (hunk 2 above)
Comment (SIGSTACKFAULT) [reply crates/bevy_gizmos/src/circles.rs:8]: Yeah i thought it was pub before. just reverted it; that can be a different PR.

File: crates/bevy_gizmos/src/circles.rs (hunk 1 above)
Comment (SIGSTACKFAULT) [reply crates/bevy_gizmos/src/circles.rs:10]: not sure why i made this public. seems to work fine when made private.
maybe I made it pub when some of the stuff I moved was still in gizmos.rs

File: crates/bevy_gizmos/src/lib.rs (hunk 3 above)
Comment (SIGSTACKFAULT) [reply crates/bevy_gizmos/src/lib.rs:20]: TIL it's possible to have a struct returned by a function which isn't exported anywhere

Comment (alice-i-cecile) [reply crates/bevy_gizmos/src/lib.rs:20]: Yeah, Rust's module privacy rules are bizarre, where you can create "sealed" traits and structs.

This got used to create unimplementable traits, so now they can't fix it.

**Comment by SIGSTACKFAULT** [comment]
Need to re-export `CircleBuilder` and `CircleBuilder2d` inside gizmos.rs, i presume?

**Review by devil-ira — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
> Need to re-export CircleBuilder and CircleBuilder2d inside gizmos.rs, i presume?

Yeah, if you want to avoid breaking changes that's the plan. Personally I think we should just make the breaking change now :)

**Review by alice-i-cecile — APPROVED** [review]

**Comment by SIGSTACKFAULT** [comment]
I mean if _you_ want it to be a breaking change that's also fine by me.

File: crates/bevy_gizmos/src/circles.rs [hunk 4]
```diff
@@ -0,0 +1,145 @@
+//! Additional [`Gizmos`] Functions -- Arrows
```
Comment (mockersf) [inline crates/bevy_gizmos/src/circles.rs:1]:
```suggestion
//! Additional [`Gizmos`] Functions -- Circles
```

**Review by mockersf — APPROVED** [review]

**Comment by SIGSTACKFAULT** [comment]
*snaps fingers* what i should've said in the first place is "I didn't plan for this to be a breaking change but i'm fine with breaking or non-breaking"

**Comment by SIGSTACKFAULT** [comment]
added migration guide to description

**Comment by SIGSTACKFAULT** [comment]
check-unused-dependencies: "failed to compile `cargo-udeps v0.1.43`".
Doesn't even get to actual dependency checking.

**Comment by SIGSTACKFAULT** [comment]
upstream https://github.com/est31/cargo-udeps/issues/228 and https://github.com/rust-lang/cargo/pull/13004

**Pull request merged**